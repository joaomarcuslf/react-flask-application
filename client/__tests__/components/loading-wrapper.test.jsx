import React, { Children } from 'react';
import renderer from 'react-test-renderer';
import LoadingWrapper from '../../app/components/loading-wrapper';

const getChildRecurr = (tree) => {
  if(tree.children) {
    return getChildRecurr(tree.children[0]);
  }

  return tree;
}

test('LoadingWrapper shows an loading text before showing the real component', () => {
  const component = renderer.create(
    <LoadingWrapper loading>
      Test
    </LoadingWrapper>,
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  expect(getChildRecurr(tree)).toEqual('Loading...');
});


test('LoadingWrapper shows the error message if it is provided', () => {
  const component = renderer.create(
    <LoadingWrapper loading={false} error={{ message: "Error" }}>
      Test
    </LoadingWrapper>,
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  expect(getChildRecurr(tree)).toEqual('Request Error.');
});

test('LoadingWrapper shows the provided child when everything is fine', () => {
  const component = renderer.create(
    <LoadingWrapper loading={false}>
      Test
    </LoadingWrapper>,
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  expect(getChildRecurr(tree)).toEqual('Test');
});
