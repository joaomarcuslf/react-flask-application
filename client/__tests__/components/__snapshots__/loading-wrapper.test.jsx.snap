// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWrapper shows an loading text before showing the real component 1`] = `
<div
  className="base__Container-sc-1fd1e17-0 daUREX"
>
  <div
    className="base__Row-sc-1fd1e17-1 hbvSyV"
    direction="column"
  >
    <h3
      className="base__Subtitle-sc-1fd1e17-4 evAyce"
    >
      Loading...
    </h3>
  </div>
</div>
`;

exports[`LoadingWrapper shows the error message if it is provided 1`] = `
<div
  className="base__Container-sc-1fd1e17-0 daUREX"
>
  <div
    className="base__Row-sc-1fd1e17-1 hbvSyV"
    direction="column"
  >
    <div
      className="base__Column-sc-1fd1e17-2 eIFjkW"
    >
      <h1
        className="base__Title-sc-1fd1e17-3 folmnf"
      >
        Request Error.
      </h1>
    </div>
    <div
      className="base__Column-sc-1fd1e17-2 eIFjkW"
    >
      <h3
        className="base__Subtitle-sc-1fd1e17-4 kyYsDJ"
      >
        Error
      </h3>
    </div>
    <div
      className="base__Column-sc-1fd1e17-2 eIFjkW"
    >
      <button
        className="form__FormButton-vl64uu-5 nOgZl"
        disabled={false}
        onClick={[Function]}
      >
        Back
      </button>
    </div>
  </div>
</div>
`;

exports[`LoadingWrapper shows the provided child when everything is fine 1`] = `"Test"`;
